src/AndcultureCode.CSharp.Sitefinity.Testing/Extensions/RestResponseResultExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using AndcultureCode.CSharp.Sitefinity.Core.Models.Services;
+using AndcultureCode.CSharp.Sitefinity.Testing.Constants;
+using Shouldly;
+
+namespace AndcultureCode.CSharp.Sitefinity.Testing.Extensions
+{
+    public static class RestResponseResultExtensions
+    {
+        public static void ShouldBeExpectedStatusCode<T>(this RestResponseResult<T> restResponseResult)
+        {
+            restResponseResult.WasExpectedStatusCode.ShouldBeTrue(
+                GetUnexpectedStatusCodeErrorMessage(restResponseResult));
+        }
+
+        public static string GetUnexpectedStatusCodeErrorMessage<T>(this RestResponseResult<T> restResponseResult)
+        {
+            var baseErrorMessage = $"Expected status code is '{restResponseResult.ExpectedStatusCode}' but actual status was code was '{restResponseResult.RestResponse.StatusCode}'.";
+            var errorMessage = $"{baseErrorMessage}'.  REST response's content is: '{restResponseResult.RestResponse.Content}'";
+
+            if (restResponseResult.RestResponse.IsWaitAMomentResponse())
+            {
+                /*
```

Comment:
Praise: Thank you for adding this comment. It provides a lot of context around Sitefinity's current behavior.